README.md

```diff
@@ -7,290 +7,505 @@
+
+The examples below assume that you followed the installation steps above and are using a `.env` file and loader to store credentials.
+
+### Login
+
+The easiest and most secure way to handle logins to a web application is to use the [Authentication Code grant](https://auth0.com/docs/api-auth/tutorials/authorization-code-grant) combined with Auth0's Universal Login page. In short, that process is:
 
-$domain        = 'YOUR_NAMESPACE';
-$client_id     = 'YOUR_CLIENT_ID';
-$client_secret = 'YOUR_CLIENT_SECRET';
-$redirect_uri  = 'http://YOUR_APP/callback';
+1. An user requesting access is redirected to the Universal Login Page.
+2. The user authenticates using one of many possible connections: social (Twitter or Facebook); database (email and password); passwordless (email or a mobile device).
+3. The user is redirected back to your application's callback URL with a `code` and `state` parameter if successful or an `error` and `error_description` if not.
+4. If the authentication was successful, the `state` parameter is validated.
+5. If the `state` is valid, the `code` parameter is exchanged with Auth0 for an access token.
+6. If the exchange is successful, the access token is used to call an Auth0 `/userinfo` endpoint, which returns the authenticated user's information.
+7. This information can be used to create an account, to start an application-specific session, or to persist as the user session.
 
+The PHP SDK handles most of the previous steps. Your application needs to:
+
+1. Determine a log in action (for example: click a link, visit walled content, etc.) and call  `Auth0::login()`
+2. Handle returned errors.
+
+A simple implementation of these steps looks like this:
+
+```php
+// Example #1
+// login.php
+use Auth0\SDK\Auth0;
+
+// Initialize the Auth0 class with required credentials.
 $auth0 = new Auth0([
-  'domain' => $domain,
-  'client_id' => $client_id,
-  'client_secret' => $client_secret,
-  'redirect_uri' => $redirect_uri,
-  'audience' => 'urn:test:api',
-  'persist_id_token' => true,
-  'persist_access_token' => true,
-  'persist_refresh_token' => true,
+
+    // See Installation above to setup environment variables.
+    'domain' => getenv('AUTH0_DOMAIN'),
+    'client_id' => getenv('AUTH0_CLIENT_ID'),
+    'client_secret' => getenv('AUTH0_CLIENT_SECRET'),
+    'audience' => getenv('AUTH0_AUTH_AUDIENCE'),
+
+    // This would be the URL for this file in this example.
+    'redirect_uri' => getenv('AUTH0_LOGIN_CALLBACK_URL'),
+
+    // The scope determines what data is provided by the /userinfo endpoint.
+    'scope' => 'openid',
 ]);
 
-$userInfo = $auth0->getUser();
+if (! empty($_GET['error']) || ! empty($_GET['error_description'])) {
+    // Handle errors sent back by Auth0.
+}
+
+// If there is a user persisted (PHP session by default), return that.
+// Otherwise, look for a `state` and `code` URL parameter to validate and exchange, respectively.
```

Comment:
All handled by the SDK behind the scenes

Reply:
It's confusing because the comment says "look for this parameters to do this other thing", but now you say that's handled internally. Better rephrase :P